File: src/main/java/de/thatsich/common/module/item/AItem.java
package de.thatsich.common.module.item;

import net.minecraft.client.renderer.texture.IIconRegister;
import net.minecraft.item.Item;

public abstract class AItem extends Item implements IItem
{
	private final AItemTexture itemTexture;

	protected AItem( AItemTexture itemTexture )
	{
		super();

		this.itemTexture = itemTexture;
	}

	@Override
	public void registerIcons ( final IIconRegister par1IconRegister )
	{
		final String texture = this.itemTexture.getTexture();
		this.itemIcon = par1IconRegister.registerIcon( texture );
	}
}

File: src/main/scala/de/thatsich/minecraft/core/module/item/AItem.scala
package de.thatsich.minecraft.core.module.item

import de.thatsich.minecraft.core.module.ATexture
import net.minecraft.client.renderer.texture.IIconRegister
import net.minecraft.item.Item

/**
 *
 *
 * @author thatsIch
 * @since 07.04.2014.
 */
abstract class AItem( itemTexture: ATexture )
	extends Item
{
	override def registerIcons( iconRegister: IIconRegister ): Unit =
	{
		val texture: String = this.itemTexture.toString
		this.itemIcon = iconRegister.registerIcon( texture )
	}
}
